typhos/display.py

```diff
@@ -80,6 +81,43 @@ def __init__(self, parent=None, **kwargs):
         self.layout().setContentsMargins(0, 0, 0, 0)
         self.load_template()
 
+        self.setContextMenuPolicy(Qt.DefaultContextMenu)
+        self.contextMenuEvent = self.open_context_menu
+
+    def generate_context_menu(self):
+        """
+        Generates the custom context menu, and populates it with any external
+        tools that have been loaded.  PyDMWidget subclasses should override
+        this method (after calling superclass implementation) to add the menu.
+
+        Returns
+        -------
+        QMenu
+        """
+        def switch_template(fname):
+            self.force_template = fname
+
+        menu = QMenu(parent=self)
+
+        for display_name, filename in self.templates.items():
+            action = menu.addAction(display_name)
+            action.triggered.connect(functools.partial(switch_template,
```

Comment:
I believe if you put the `switch_template` in the `for` loop, you can use `filename` as-is:

```suggestion
            def switch_template():
                self.force_template = filename
            action.triggered.connect(switch_template)
```